### PR TITLE
feat: Dark overlay on pending & failed MFM.

### PR DESCRIPTION
**Pull request by liamcho**
Dark overlay on pending & failed MFM: 
- Dynamically remove shade from images that has been uploaded.
- Right now uploaded files are being displays even before upload. Fix this to display only when upload has completed. If not uploaded, just display default.
- Need to implement pending display in message thread (child MFM)
- Fix `useSendMultipleFilesMessage.spec.ts` test. See below image why this fails.

Screen Shot 2023-09-27 at 3 38 40 PM

Reelia-recording-2023-09-27
Reelia-recording-2023-09-27 (1)

Fixes: [UIKIT-4510](https://sendbird.atlassian.net/browse/UIKIT-4510)

[UIKIT-4510]: https://sendbird.atlassian.net/browse/UIKIT-4510?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/ui/MultipleFilesMessageItemBody/useDynamicUploadedIndex.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  useEffect(() => {
+    if (
+      message?.sendingStatus !== SendingStatus.SUCCEEDED
+      && Array.isArray(message['statefulFileInfoList'])
+    ) {
+      const newStatefulFileInfoList: StatefulFileInfo[] = createStatefulFileInfoList(message, statefulFileInfoList);
+      setStatefulFileInfoList(newStatefulFileInfoList);
+    }
+  }, [message['statefulFileInfoList']?.map((fileInfo: StatefulFileInfo) => fileInfo.isUploaded).join(',')]);
```
Comment (HoonBaek) [inline src/ui/MultipleFilesMessageItemBody/useDynamicUploadedIndex.ts:18]: It is expected that the timing of changes to the `message.statefulFileInfoList` aligns with the timing of changes to each `fileInfo.isUploaded`.

Therefore, it seems that making the array to string using the .`join` in the code is unnecessary.

File: src/utils/createStatefulFileInfoList.ts
Comment (HoonBaek) [inline src/utils/createStatefulFileInfoList.ts:1]: 👍
good to add a test for this, but we can add it later too

**Review by HoonBaek — COMMENTED** [review]
Also, pls check the failed test 🙏

File: src/ui/FileViewer/index.tsx [hunk 2]
```diff
@@ -126,7 +126,7 @@ export const FileViewerComponent = (props: FileViewerComponentProps): ReactEleme
 };
 
 export interface FileViewerProps {
-  message: FileMessage | MultipleFilesMessage;
+  message?: FileMessage | MultipleFilesMessage;
```
[the remark this answers is not in the document]
Comment (liamcho) [reply src/ui/FileViewer/index.tsx:129]: honestly I can't remember right now. But im guessing there was a case when message is undefined.

File: src/ui/ImageRenderer/index.tsx [hunk 3]
```diff
@@ -144,11 +146,12 @@ const ImageRenderer = ({
       {HiddenImageLoader}
       {
         shadeOnHover && <div
-        className="sendbird-multiple-files-image-renderer__image-cover"
-        style={{
-          borderRadius: getBorderRadiusForImageRenderer(circle, borderRadius),
```
[the remark this answers is not in the document]
Comment (liamcho) [reply src/ui/ImageRenderer/index.tsx:149]: so what I want to do is:
- always add borderRadius
- optionally add display

is there a better way to achieve this?

File: src/ui/MultipleFilesMessageItemBody/index.tsx [hunk 4]
```diff
@@ -39,37 +41,37 @@ export default function MultipleFilesMessageItemBody({
   threadMessageKind,
 }: Props): ReactElement {
   const { isMobile } = useMediaQueryContext();
-  const [fileInfoList] = useState<UploadedFileInfo[]>(message.fileInfoList);
   const threadMessageKindKey = useThreadMessageKindKeySelector({ threadMessageKind, isMobile });
-  const [currentIndex, setCurrentIndex] = useState(-1);
+  const [currentFileViewerIndex, setCurrentFileViewerIndex] = useState(-1);
+  const updatedFileInfoList = useDynamicUploadedIndex(message);
```
[the remark this answers is not in the document]
Comment (liamcho) [reply src/ui/MultipleFilesMessageItemBody/index.tsx:46]: ok ill do that but

This ui/MultipleFilesMessageItemBody component is an UI component, so it's good to not use our custom hook inside of it.

^ why is this the case?

File: src/ui/MultipleFilesMessageItemBody/useDynamicUploadedIndex.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from 'react';
+import { MultipleFilesMessage, SendingStatus } from '@sendbird/chat/message';
+import { createStatefulFileInfoList, StatefulFileInfo } from '../../utils/createStatefulFileInfoList';
+
+export const useDynamicUploadedIndex = (message: MultipleFilesMessage): StatefulFileInfo[] => {
```
[the remark this answers is not in the document]
Comment (liamcho) [reply src/ui/MultipleFilesMessageItemBody/useDynamicUploadedIndex.ts:5]: ah yes of course i forgot to change this name

File: src/ui/MultipleFilesMessageItemBody/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (HoonBaek) [reply src/ui/MultipleFilesMessageItemBody/index.tsx:46]: If UI components are separated into another repo, It won't be good to use it and not simple to use

**Comment by liamcho** [comment]
> Also, pls check the failed test 🙏

I need your help on this. Can we check tomorrow together?

File: src/utils/createStatefulFileInfoList.ts
Comment (liamcho) [reply src/utils/createStatefulFileInfoList.ts:1]: I agree but let me do this in separate PR after resolving all current issues lined up.

**Comment by liamcho** [comment]
So the reason why tests were failing is because test was mocking send message. It cannot verify the order of which the events are being called by SDK. We need to fix the test to verify the order of actual events. @HoonBaek let's consider this a separate task.

**Review by HoonBaek — APPROVED** [review]
LGTM, thanks for this awesome job!

**Comment by liamcho** [comment]
@HoonBaek I applied new SDK version in the latest commit. please review again, thanks 😉

**Review by HoonBaek — APPROVED** [review]
Thank you so much for doing such a wonderful job!

**Pull request merged**